src/main/java/software/amazon/nio/spi/s3/S3FileSystemProvider.java

```diff
@@ -1142,4 +1133,12 @@ private S3Path forceAwsClient(final Path path, final S3AsyncClient client) {
         p.getFileSystem().clientProvider(new FixedS3ClientProvider(client));
         return p;
     }
+
+    protected static S3Path checkPath(Path obj) {
+        if (obj == null)
```

Comment:
You could use `Objects.requireNotNull(obj)`

Reply:
Changed in 62b7f74